File: MiniDOMFinanziaria/Finanziaria/EstinzioniXEstintore/EstinzioneXEstintoreCursor.cs
using System;

namespace minidom
{
    public partial class Finanziaria
    {

        /// <summary>
    /// Cursore sulla tabella delle estinzioni
    /// </summary>
    /// <remarks></remarks>
        public class EstinzioneXEstintoreCursor : Databases.DBObjectCursor<EstinzioneXEstintore>
        {
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<bool> m_Selezionata = new DBCursorField<bool>("Selezionata");
            private DBCursorField<int> m_IDEstinzione = new DBCursorField<int>("IDEstinzione");
            private DBCursorField<int> m_IDEstintore = new DBCursorField<int>("IDEstintore");
            private DBCursorStringField m_TipoEstintore = new DBCursorStringField("TipoEstintore");
            private DBCursorField<DateTime> m_DataDecorrenza = new DBCursorField<DateTime>("DataDecorrenza");
            private DBCursorField<int> m_NumeroQuoteInsolute = new DBCursorField<int>("NQI");
            private DBCursorField<int> m_NumeroQuoteScadute = new DBCursorField<int>("NQS");
            private DBCursorField<int> m_NumeroQuoteResidue = new DBCursorField<int>("NQR");
            private DBCursorStringField m_Parametro = new DBCursorStringField("Parametro");
            private DBCursorField<decimal> m_Correzione = new DBCursorField<decimal>("Correzione");
            private DBCursorField<double> m_PenaleEstinzione = new DBCursorField<double>("PenaleEstinzione");
            private DBCursorField<DateTime> m_DataEstinzione = new DBCursorField<DateTime>("DataEstinzione");
            private DBCursorField<decimal> m_Rata = new DBCursorField<decimal>("Rata");
            private DBCursorField<DateTime> m_DataFine = new DBCursorField<DateTime>("DataFine");
            private DBCursorField<int> m_Durata = new DBCursorField<int>("Durata");
            private DBCursorField<double> m_TAN = new DBCursorField<double>("TAN");
            private DBCursorField<double> m_TAEG = new DBCursorField<double>("TAEG");
            private DBCursorField<double> m_TotaleDaEstinguere = new DBCursorField<double>("TotaleDaEstinguere");
            private DBCursorField<int> m_IDCessionario = new DBCursorField<int>("IDCessionario");
            private DBCursorStringField m_NomeCessionario = new DBCursorStringField("NomeCessionario");
            private DBCursorStringField m_NomeAgenzia = new DBCursorStringField("NomeAgenzia");
            private DBCursorField<TipoEstinzione> m_Tipo = new DBCursorField<TipoEstinzione>("Tipo");
            private DBCursorField<decimal> m_MontanteResiduo = new DBCursorField<decimal>("MontanteResiduo");
            private DBCursorField<DateTime> m_DataCaricamento = new DBCursorField<DateTime>("DataCaricamento");
            private DBCursorField<decimal> m_MontanteResiduoForzato = new DBCursorField<decimal>("MontanteResiduoForzato");

            public EstinzioneXEstintoreCursor()
            {
            }

            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            public DBCursorField<decimal> MontanteResiduoForzato
            {
                get
                {
                    return m_MontanteResiduoForzato;
                }
            }

            public DBCursorField<DateTime> DataCaricamento
            {
                get
                {
                    return m_DataCaricamento;
                }
            }

            public DBCursorField<double> PenaleEstinzione
            {
                get
                {
                    return m_PenaleEstinzione;
                }
            }

            public DBCursorField<decimal> MontanteResiduo
            {
                get
                {
                    return m_MontanteResiduo;
                }
            }

            public DBCursorField<TipoEstinzione> Tipo
            {
                get
                {
                    return m_Tipo;
                }
            }

            public DBCursorField<int> IDCessionario
            {
                get
                {
                    return m_IDCessionario;
                }
            }

            public DBCursorStringField NomeCessionario
            {
                get
                {
                    return m_NomeCessionario;
                }
            }

            public DBCursorStringField NomeAgenzia
            {
                get
                {
                    return m_NomeAgenzia;
                }
            }

            public DBCursorField<DateTime> DataEstinzione
            {
                get
                {
                    return m_DataEstinzione;
                }
            }

            public DBCursorField<decimal> Rata
            {
                get
                {
                    return m_Rata;
                }
            }

            public DBCursorField<DateTime> DataFine
            {
                get
                {
                    return m_DataFine;
                }
            }

            public DBCursorField<int> Durata
            {
                get
                {
                    return m_Durata;
                }
            }

            public DBCursorField<double> TAN
            {
                get
                {
                    return m_TAN;
                }
            }

            public DBCursorField<double> TAEG
            {
                get
                {
                    return m_TAEG;
                }
            }

            public DBCursorField<double> TotaleDaEstinguere
            {
                get
                {
                    return m_TotaleDaEstinguere;
                }
            }

            public DBCursorField<bool> Selezionata
            {
                get
                {
                    return m_Selezionata;
                }
            }

            public DBCursorStringField Parametro
            {
                get
                {
                    return m_Parametro;
                }
            }

            public DBCursorField<int> IDEstinzione
            {
                get
                {
                    return m_IDEstinzione;
                }
            }

            public DBCursorField<int> IDEstintore
            {
                get
                {
                    return m_IDEstintore;
                }
            }

            public DBCursorStringField TipoEstintore
            {
                get
                {
                    return m_TipoEstintore;
                }
            }

            public DBCursorField<DateTime> DataDecorrenza
            {
                get
                {
                    return m_DataDecorrenza;
                }
            }

            public DBCursorField<int> NumeroQuoteInsolute
            {
                get
                {
                    return m_NumeroQuoteInsolute;
                }
            }

            public DBCursorField<int> NumeroQuoteScadute
            {
                get
                {
                    return m_NumeroQuoteScadute;
                }
            }

            public DBCursorField<int> NumeroQuoteResidue
            {
                get
                {
                    return m_NumeroQuoteResidue;
                }
            }

            public DBCursorField<decimal> Correzione
            {
                get
                {
                    return m_Correzione;
                }
            }

            public override object InstantiateNew(DBReader dbRis)
            {
                return new EstinzioneXEstintore();
            }

            public override string GetTableName()
            {
                return "tbl_EstinzioniXEstintore";
            }

            protected override Sistema.CModule GetModule()
            {
                return null;
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Database;
            }
        }
    }
}

File: MiniDOMFinanziaria/Finanziaria/EstinzioniXEstintore/EstinzioneXEstintoreCursor.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Public Class Finanziaria

    ''' <summary>
    ''' Cursore sulla tabella delle estinzioni
    ''' </summary>
    ''' <remarks></remarks>
    Public Class EstinzioneXEstintoreCursor
        Inherits DBObjectCursor(Of EstinzioneXEstintore)

        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_Selezionata As New CCursorField(Of Boolean)("Selezionata")
        Private m_IDEstinzione As New CCursorField(Of Integer)("IDEstinzione")
        Private m_IDEstintore As New CCursorField(Of Integer)("IDEstintore")
        Private m_TipoEstintore As New CCursorFieldObj(Of String)("TipoEstintore")
        Private m_DataDecorrenza As New CCursorField(Of Date)("DataDecorrenza")
        Private m_NumeroQuoteInsolute As New CCursorField(Of Integer)("NQI")
        Private m_NumeroQuoteScadute As New CCursorField(Of Integer)("NQS")
        Private m_NumeroQuoteResidue As New CCursorField(Of Integer)("NQR")
        Private m_Parametro As New CCursorFieldObj(Of String)("Parametro")
        Private m_Correzione As New CCursorField(Of Decimal)("Correzione")
        Private m_PenaleEstinzione As New CCursorField(Of Double)("PenaleEstinzione")
        Private m_DataEstinzione As New CCursorField(Of Date)("DataEstinzione")
        Private m_Rata As New CCursorField(Of Decimal)("Rata")
        Private m_DataFine As New CCursorField(Of Date)("DataFine")
        Private m_Durata As New CCursorField(Of Integer)("Durata")
        Private m_TAN As New CCursorField(Of Double)("TAN")
        Private m_TAEG As New CCursorField(Of Double)("TAEG")
        Private m_TotaleDaEstinguere As New CCursorField(Of Double)("TotaleDaEstinguere")
        Private m_IDCessionario As New CCursorField(Of Integer)("IDCessionario")
        Private m_NomeCessionario As New CCursorFieldObj(Of String)("NomeCessionario")
        Private m_NomeAgenzia As New CCursorFieldObj(Of String)("NomeAgenzia")
        Private m_Tipo As New CCursorField(Of TipoEstinzione)("Tipo")
        Private m_MontanteResiduo As New CCursorField(Of Decimal)("MontanteResiduo")
        Private m_DataCaricamento As New CCursorField(Of Date)("DataCaricamento")
        Private m_MontanteResiduoForzato As New CCursorField(Of Decimal)("MontanteResiduoForzato")

        Public Sub New()
        End Sub

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property

        Public ReadOnly Property MontanteResiduoForzato As CCursorField(Of Decimal)
            Get
                Return Me.m_MontanteResiduoForzato
            End Get
        End Property

        Public ReadOnly Property DataCaricamento As CCursorField(Of Date)
            Get
                Return Me.m_DataCaricamento
            End Get
        End Property

        Public ReadOnly Property PenaleEstinzione As CCursorField(Of Double)
            Get
                Return Me.m_PenaleEstinzione
            End Get
        End Property

        Public ReadOnly Property MontanteResiduo As CCursorField(Of Decimal)
            Get
                Return Me.m_MontanteResiduo
            End Get
        End Property

        Public ReadOnly Property Tipo As CCursorField(Of TipoEstinzione)
            Get
                Return Me.m_Tipo
            End Get
        End Property

        Public ReadOnly Property IDCessionario As CCursorField(Of Integer)
            Get
                Return Me.m_IDCessionario
            End Get
        End Property

        Public ReadOnly Property NomeCessionario As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCessionario
            End Get
        End Property

        Public ReadOnly Property NomeAgenzia As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeAgenzia
            End Get
        End Property


        Public ReadOnly Property DataEstinzione As CCursorField(Of Date)
            Get
                Return Me.m_DataEstinzione
            End Get
        End Property

        Public ReadOnly Property Rata As CCursorField(Of Decimal)
            Get
                Return Me.m_Rata
            End Get
        End Property

        Public ReadOnly Property DataFine As CCursorField(Of Date)
            Get
                Return Me.m_DataFine
            End Get
        End Property

        Public ReadOnly Property Durata As CCursorField(Of Integer)
            Get
                Return Me.m_Durata
            End Get
        End Property

        Public ReadOnly Property TAN As CCursorField(Of Double)
            Get
                Return Me.m_TAN
            End Get
        End Property

        Public ReadOnly Property TAEG As CCursorField(Of Double)
            Get
                Return Me.m_TAEG
            End Get
        End Property

        Public ReadOnly Property TotaleDaEstinguere As CCursorField(Of Double)
            Get
                Return Me.m_TotaleDaEstinguere
            End Get
        End Property

        Public ReadOnly Property Selezionata As CCursorField(Of Boolean)
            Get
                Return Me.m_Selezionata
            End Get
        End Property

        Public ReadOnly Property Parametro As CCursorFieldObj(Of String)
            Get
                Return Me.m_Parametro
            End Get
        End Property

        Public ReadOnly Property IDEstinzione As CCursorField(Of Integer)
            Get
                Return Me.m_IDEstinzione
            End Get
        End Property

        Public ReadOnly Property IDEstintore As CCursorField(Of Integer)
            Get
                Return Me.m_IDEstintore
            End Get
        End Property

        Public ReadOnly Property TipoEstintore As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoEstintore
            End Get
        End Property

        Public ReadOnly Property DataDecorrenza As CCursorField(Of Date)
            Get
                Return Me.m_DataDecorrenza
            End Get
        End Property

        Public ReadOnly Property NumeroQuoteInsolute As CCursorField(Of Integer)
            Get
                Return Me.m_NumeroQuoteInsolute
            End Get
        End Property

        Public ReadOnly Property NumeroQuoteScadute As CCursorField(Of Integer)
            Get
                Return Me.m_NumeroQuoteScadute
            End Get
        End Property

        Public ReadOnly Property NumeroQuoteResidue As CCursorField(Of Integer)
            Get
                Return Me.m_NumeroQuoteResidue
            End Get
        End Property

        Public ReadOnly Property Correzione As CCursorField(Of Decimal)
            Get
                Return Me.m_Correzione
            End Get
        End Property

        Public Overrides Function InstantiateNew(dbRis As DBReader) As Object
            Return New EstinzioneXEstintore
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_EstinzioniXEstintore"
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Finanziaria.Database
        End Function



    End Class


End Class
